### DISCUSSION TITLE
Feedback & Stories & Survey

### DISCUSSION THREAD

**Discussion by palkan** [anycable/anycable#68] [General]
Hey everyone!

First, here is the survey for AnyCable users: https://evilmartians.typeform.com/to/W35uRm.
We want to learn more about you to make this piece of software better.

This thread (not a an issue at all) aims to collect your feedback/stories in a free form:
- “How and why I switched to AnyCable”
- “How AnyCable helped us to improve the app performance”
- “Why we migrated from AnyCable to X” (yep, un-success stories are welcomed too)

That would help others to decide whether AnyCable is the right tool or not, and help us to improve it.

Please, comment on this issue to share your story!

**Comment by pokonski** [comment]
Hey!

I am developing a time tracking tool that lets people collaborate on projects, AnyCable is used to deliver updates to other members in real time. Previously it was done via polling but that proved to be too stressful on the server so we looked for alternatives.

In recent years we grew a lot so ActionCable started consuming way too much resources (exactly like the original benchmark said here https://evilmartians.com/chronicles/anycable-actioncable-on-steroids ), so I looked into alternatives. 

I used Anycable-Go for the WS server and it turned out fantastically, it's basically zero maintenance and is working in production for months without any hiccups! 

The migration was very simple, just one additional service to start on the server, it's still using Redis so no other architectural changes were needed. Also our codebase did not require any changes in ActionCable channels or connection logic at all :boom:

**Comment by Vasfed** [comment]
For realtime notifications in our application we used Faye (to be exact - PrivatePub inside thin with redis backend) for quite some time, apart from crashing sometimes and being not-so-easy to monitor - there were no big issues with it.
After ActionCable was released I really liked concept of abstracting pub-sub topics, also long-polling fallbacks of faye seemed outdated, so upgrade was put on backlog for after rails 5 migration.

Since production is running on unicorn - anycable was easier to migrate to than vanilla actioncable, as it did not require any modifications to http stack, while also keeping benefits of separate server for websockets. Did not run serious load tests, but CPU load and message latency have decreased, so I'm pretty happy with the decision.

**Comment by palkan** [comment]
Hey @pokonski and @Vasfed!

Thanks for sharing your stories! Could you also please fill a little AnyCable survey: 
https://evilmartians.typeform.com/to/W35uRm ?

**Comment by proudlygeek** [comment]
Hey 👋 

I'm developing a commercial SaaS product, a shared inbox for Facebook Messenger and WhatsApp for teams, which started as a Rails app with ActionCable; as soon as we hit 200 concurrent connection we started to have lots of errors and failed connections using plain ActionCable, so we started considering to rewrite our chat module using using Elixir / Phoenix or Golang. 

Fortunately we found out about AnyCable 💖

We started using AnyCable since 8-9 months and it has been working great! We're handling at the moment ~1000k concurrent connections on a single Heroku Dyno with no issues, setup and docs are clear and the actual migration itself from ActionCable was really quick and painless. 💪 🔥 

I'm really happy about the choice, no doubts! 💯
